app/Controllers/Page.php

```diff
@@ -90,6 +92,106 @@ public function photoCredits()
         return $photos;
     }
 
+    /*
+    Create a new Wordpress Query for looping `pcc-story` post types.
+    */
+    public function storiesQuery ()
+    {
+        $meta_query_args = array();
+
+        /* If the clear parameter is set, unset all parameters so they
+        aren't queried. */
+        if (get_query_var( 'clear' )) {
+
+            /* TODO: Improve this implementation. Parameters should
+            be unset on submission and before reaching this point. */
+            remove_query_arg ( 'org' );
+            remove_query_arg( 'clear' );
+        } else if (get_query_var( 'org' )) {
+            /* If filtering by a single organization name. */
+            $meta_query_args = [
+              'key' => 'pcc_story_organization',
+              'value' => get_query_var( 'org' )
+            ];
+        }
+
+        $query = new \WP_Query(
+            [
+                'post_type' => 'pcc-story',
+                'posts_per_page' => -1,
+                'post__in' => $postIds,
+                'orderby' => 'post_date',
+                'order' => 'desc',
+                'meta_query' => array( $meta_query_args ),
+            ]
+        );
+        return $query;
+    }
+
+    /*
+    Return an array of unique organization names sorted alphabetically,
+    false if there are no organizations.
+    */
+    public function storyOrgs()
+    {
+      $terms = get_terms ( 'pcc-organization' );
+
+      if ($terms && ! is_wp_error( $terms ) ) {
+
+          foreach ($terms as $term ) {
+              $results[] = $term->name;
+          }
+
+          $results = array_unique( $results );
+          sort( $results );
+
+          return $results;
+      }
+
+      return false;
+    }
+
+    /*
+    Create a link list for navigating Stories by taxonomy terms. The first link
+    of the list is a link to the Community Stories page which lists all stories.
+    Returns false if there are no terms, or the supplied taxonomy name is
+    invalid.
+    */
+    public static function taxonomy_menu_list ( $taxonomy = false )
+    {
+        if ( $taxonomy ) {
+            $terms = get_terms ( $taxonomy );
+
+            if ($terms && ! is_wp_error( $terms ) ) {
+                $li_class = 'link-list__item';
+                $output .= '<ul class="link-list">';
```

Comment:
This should be:

```suggestion
                $output = '<ul class="link-list">';
```

The `$output` variable hasn't been defined yet. Gives me a warning with `WP_DEBUG` on.